From: Carlos Garnacho <carlosg@gnome.org>
Date: Sat, 14 Jul 2018 00:14:07 +0200
Subject: keyboard: defer position-changed till we have a rect

Emitting it that soon results in JS warnings, as we don't have
everything in place yet. The position-changed signal will be
emitted from other locations as soon as we have it.

Bug: https://gitlab.gnome.org/GNOME/gnome-shell/issues/464
Origin: upstream, 3.29.91, commit:81956e9b84a751e0aae739ef08017d3bfef907c8
---
 js/ui/keyboard.js | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/js/ui/keyboard.js b/js/ui/keyboard.js
index 1666a50..4a24940 100644
--- a/js/ui/keyboard.js
+++ b/js/ui/keyboard.js
@@ -526,9 +526,7 @@ var FocusTracker = new Lang.Class({
         this._currentWindow = window;
         if (window) {
             this._currentWindowPositionId = this._currentWindow.connect('position-changed', () => {
-                if (global.display.get_grab_op() == Meta.GrabOp.NONE)
-                    this.emit('position-changed');
-                else
+                if (global.display.get_grab_op() != Meta.GrabOp.NONE)
                     this.emit('reset');
             });
         }
